docs/sources/v0.51.x/using-k6-browser/migrating-to-k6-v0-52.md

```diff
@@ -0,0 +1,215 @@
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  await page.$('#text1').then((text) => text.type('hello world'));
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+Instead you could do this until k6 `v0.52` is released:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const text = await page.$('#text1');
+  await text.type('hello world');
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+## Working with k6 check
+
+The k6 `check` API wasn't designed to work with async APIs. However there is a workaround. In your current scripts, you may use a `check` like so:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  check(page, {
+    header: (p) => p.locator('h2').textContent() == 'Welcome, admin!',
+  });
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+Since the `locator.textContent` API is being migrated to async, and `check` doesn't work with async APIs, you will need to do something like this:
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import { check } from 'k6';
+
+export default async function () {
+  const page = browser.newPage();
+
+  // ...
+
+  const h2 = page.locator('h2');
+  const headerOK = (await h2.textContent()) == 'Welcome, admin!';
+  check(headerOK, { header: headerOK });
+
+  // ...
+}
+```
+
+{{< /code >}}
+
+## Groups
+
+A note on groups, they do not work with async APIs either, there is no workaround as of yet. Here's the [github issue](https://github.com/grafana/k6/issues/2728) that you can follow to keep up to date with relevant news on a groups API that works with async APIs.
+
+# Are there any other upcoming major breaking changes?
+
+We think that this is likely the last large breaking change until the browser module is deemed to be stable enough to be a non-experimental module in k6. When the browser module is no longer experimental, we think it should only be a small change in the import path from `k6/experimental/browser` to `k6/browser`. For more information on graduating extension out of experimental, refer to the [extension graduation process]({{< relref "../extensions/explanations/extension-graduation" >}}).
+
+# Where to get extra help
+
+We will also be releasing the latest type definitions and updating the documentation when we release k6 `v0.52`. Refer to the [k6 IntelliSense]({{< relref "../set-up/configure-k6-intellisense" >}}) documentation on how to work with the type definition file in your project.
+
+For all the details, make sure to review the **upcoming** changelog for [k6 version 0.52](https://github.com/grafana/k6/releases/tag/v0.52.0). As always, ask in [the community forum](https://community.grafana.com/c/grafana-k6/k6-browser/79) if you need help!
```

Comment:
Should I remove the link to the missing release notes file for now?

Reply:
Yeah, I wouldn't add a missing link here. I think we could change it to:

```suggestion
For all the details, make sure to review the _upcoming_ CHANGELOG for [k6 version 0.52](https://github.com/grafana/k6/releases/). You can also ask questions in [the community forum](https://community.grafana.com/c/grafana-k6/k6-browser/79) if you need help.
```

Reply:
Thanks! I've added this suggestion and also all the suggestions from https://github.com/grafana/k6-docs/pull/1606